### DISCUSSION TITLE
Are there plans for `bun` to read scripts from `stdin` like `node`?

### DISCUSSION THREAD

**Discussion by winstxnhdw** [oven-sh/bun#4484] [Ideas]
I tried piping a script into `bun`, but it seems it doesn't accept `stdin`.

```
❯ node --help
Usage: node [options] [ script.js ] [arguments]
       node inspect [options] [ script.js | host:port ] [arguments]

Options:
  -         script read from stdin (default if no file name is provided, interactive mode if a tty)
  - ...
```

**Comment by orl0** [comment]
Upvote!

Also I've created a simple script to add `stdin` support.

1. Get [this shell script](https://gist.github.com/orl0/bce9b6747eb92a386d11d51b487372cb)
2. `chmod +x bun.sh`
3. Place it somewhere within your `$PATH`
4. Create `alias bun='bun.sh'`
5. Use it as `echo 'console.log("test")' | bun`

**Reply by winstxnhdw** [reply]
@orl0 this kinda defeats the purpose of piping to `stdin` if there's an intermediary step of writing to disk behind the scenes. the whole point of using a pipe is to run everything in-memory and without more than necessary syscalls. this usually translates to a miniscule performance win in CI.
